platform/lib/routers/pages.js

```diff
@@ -144,16 +147,36 @@ if (config.environment === 'development') {
 }
 
 if (config.environment !== 'development') {
-  pages.use('/', (request, response, next) => {
-    // Check if this request should be filtered
+  const STATIC_PAGES_PATH = utils.project.absolute('platform/pages');
+  const staticMiddleware = express.static(STATIC_PAGES_PATH);
+
+  pages.use('/', async (request, response, next) => {
+    let requestPath = request.path;
     const activeFormat = getFilteredFormat(request);
-    if (activeFormat) {
-      // And if it should be filtered rewrite to the correct file
-      request.url = request.url.replace('.html', `.${activeFormat}.html`);
-    }
 
-    next();
-  }, express.static('pages'));
+    // Only handle filtered requests on our own ...
+    // eslint-disable-next-line max-len
+    if (activeFormat && isFilterableRoute(requestPath) && fs.existsSync(utils.project.pagePath(requestPath))) {
+      // Match root requests to a possible index.html
+      if (requestPath.endsWith('/')) {
+        requestPath = requestPath + 'index.html';
+      }
+
+      // Read the file to a variable to modify it
+      const page = await readFileAsync(utils.project.pagePath(requestPath))
+          .catch((e) => {
```

Comment:
use try/catch with async/await

Reply:
Done, but does catching here make sense at all since we are just rethrowing for the express error handler to catch?